src/ImageDetector/index.js

```diff
@@ -0,0 +1,162 @@
+import * as tf from '@tensorflow/tfjs';
+import handleArguments from '../utils/handleArguments';
+import { mediaReady } from '../utils/imageUtilities';
+
+/**
+ * @typedef {Object} SpecificDetectorImplementation
+ *
+ * @property {Promise<boolean>} ready - lets the parent detector know that the
+ * specific implementation is ready.
```

Comment:
I want to change this to be a method (the `this.loadModel` method) rather than a property because a property doesn't work as nicely when using the `@implements {SpecificDetectorImplementation}` tag on the model class. Not sure if it's different in VSCode but JetBrains is warning me about the unused property and does not pick up that it implements anything.
![image](https://github.com/ml5js/ml5-next-gen/assets/28965286/d33015c9-5a70-45c9-accd-71687be2b9c3)
![image](https://github.com/ml5js/ml5-next-gen/assets/28965286/25e85514-40b0-4709-8291-aa60acee2b9d)